### PR TITLE
Adding Ossm certs

### PR DESCRIPTION
**Pull request by neal-timpe**
https://issues.redhat.com/browse/OSSMDOC-14

These should be cherrypicked to 4.3-4.6, when it's finished.

### CODE REVIEW

File: modules/ossm-security-cert-manage.adoc [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+.Prerequisite
+
+* You must have installed {ProductName} with mutual TLS enabled to configure certificates.
+* Deploy the Bookinfo sample application to verify the results.
+
+== Adding an existing certificate and key
+
+To us an existing signing (CA) certificate and key, you must create a chain of trust file that includes the CA certificate, key, and root certificate. You must use the following exact file names for each of the corresponding certificates. The CA certificate is called `ca-cert.pem`, the key is `ca-key.pem`, and the root certificate, which signs `ca-cert.pem`, is called `root-cert.pem`. If your workload uses intermediate certificates, you must specify them in a `cert-chain.pem` file.
```
Comment (brian-avery) [inline modules/ossm-security-cert-manage.adoc:19]: Should this be:

To us an existing signing certificate authority (CA) certificate and key,

File: modules/ossm-security-cert-manage.adoc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    --from-file=samples/certs/cert-chain.pem
+----
++
+2. Redeploy {ProductShortName} with `global.mtls.enabled` set to `true` and `security.selfSigned` set to `false`. {ProductShortName} reads the certificates and key from the secret-mount files.
++
+----
+$ istioctl manifest apply --set values.global.mtls.enabled=true,values.security.selfSigned=false
+----
++
+3. To make sure the workloads obtain the new certificates promptly, delete the secrets generated by {ProductShortName}, named `istio.\*`. In this example, `istio.default`. {ProductShortName} issues new certificates for the workloads.
```
Comment (brian-avery) [inline modules/ossm-security-cert-manage.adoc:37]: I think this should be `istio.*` instead of `istio.\*`

File: modules/ossm-security-cert-manage.adoc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+== Verifying your certificates
+
+Use the Bookinfo sample application to verify your certificates are mounted correctly.
+
+1. Retrieve the mounted certificates. In the following, we take the ratings pod as an example, and verify the certificates mounted on the pod. Set the pod name to RATINGSPOD:
```
Comment (brian-avery) [inline modules/ossm-security-cert-manage.adoc:43]: This is storing the pods name in the variable RATINGSPOD. It's not setting the pod name.

File: modules/ossm-security-cert-manage.adoc (hunk 1 above)
Comment (brian-avery) [inline modules/ossm-security-cert-manage.adoc:19]: "To use" instead of "to us"

**Review by brian-avery — APPROVED** [review]

File: modules/ossm-security-cert-manage.adoc [hunk 4]
```diff
@@ -0,0 +1,129 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/service_mesh_user_guide/ossm-security.adoc
+
+[id="ossm-cert-manage"]
+= Adding an external certificate authority key and certificate
+
+By default, {ProductName} generates self-signed root certificate and key, and uses them to sign the workload certificates. You can also use the operator-specified certificate and key to sign workload certificates, with operator-specified root certificate. This task demonstrates an example to plug certificates and key into {ProductShortName}.
+
+.Prerequisites
```
Comment (tvieira) [inline modules/ossm-security-cert-manage.adoc:10]: The user would have to have her own certificates as well, it would be work having this info in the ```Prerequisites```

Comment (tvieira) [reply modules/ossm-security-cert-manage.adoc:10]: +1 @neal-timpe

**Comment by tvieira** [comment]
Here's my thoughts - if you are using the certificates from samples/certs/ within the istio source code (https://github.com/maistra/istio/tree/maistra-2.0/samples/certs) for the verification steps, I would then recommend you to rewrite:

- on the requirements, along with "you have the certificates" add something like, "In the instructions below we used the samples/certs certificates from the Istio source code found at ......", or something like that.
- on the command below, if you decide to go with Istio samples, then replace ```<path>``` with the same ```samples/certs``` path used the the verification steps.

```
oc create secret generic cacerts -n istio-system --from-file=<path>/ca-cert.pem \
    --from-file=<path>/ca-key.pem --from-file=<path>/root-cert.pem \
    --from-file=<path>/cert-chain.pem
```

If you don't want to make use of the ```samples/certs ```, then, on the verification steps, use ```<path>``` instead of ```samples/certs/```

I vote to use samples and explicitly tells the user where they can find these samples, so a practical exercise can be done quickly in case the user wants to test it.

I leave it to you @neal-timpe to decide, either way works IMHO. Having ```<path>``` in the first section and ```samples/certs``` in the second looks something is disconnected.

**Comment by neal-timpe** [comment]
@tvieira When I did this, I used those certs, but if I leave `samples/certs` there, the command isn't copyable, and it's not clear that you need to replace the path with the path to your local cert file. I've added the path change and an instruction to get the certs from the maistra repo. If this looks good, can you click approve?

**Review by tvieira — APPROVED** [review]
lgtm

**Review by gbaufake — APPROVED** [review]
I was able to reproduce the commands with @tvieira instructions.

LGTM

File: modules/ossm-security-cert-manage.adoc [hunk 5]
```diff
@@ -0,0 +1,130 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/service_mesh_user_guide/ossm-security.adoc
+
+[id="ossm-cert-manage"]
+= Adding an external certificate authority key and certificate
+
+By default, {ProductName} generates self-signed root certificate and key, and uses them to sign the workload certificates. You can also use the operator-specified certificate and key to sign workload certificates, with operator-specified root certificate. This task demonstrates an example to plug certificates and key into {ProductShortName}.
```
Comment (JStickler) [inline modules/ossm-security-cert-manage.adoc:8]: Capitalize Operator.

File: modules/ossm-security-cert-manage.adoc [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+.Prerequisites
+
+* You must have installed {ProductName} with mutual TLS enabled to configure certificates.
+* You must use the certificates from link:https://github.com/maistra/istio/tree/maistra-2.0/samples/certs[Maistra repo].
```
Comment (JStickler) [inline modules/ossm-security-cert-manage.adoc:13]: This makes it sound like you must use the Maistra certificates for production use, when I think you're just saying you need them for this example.  Maybe rephrase something like "The following example requires that you use the certificates from the Maistra repo.  In production you would use [other source - their own certificates?].

File: modules/ossm-security-cert-manage.adoc [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+1. Create a secret `cacert` that includes the input files `ca-cert.pem`, `ca-key.pem`, `root-cert.pem` and `cert-chain.pem`.
++
+----
+$ oc create secret generic cacerts -n istio-system --from-file=<path>/ca-cert.pem \
+    --from-file=<path>/ca-key.pem --from-file=<path>/root-cert.pem \
+    --from-file=<path>/cert-chain.pem
+----
++
+2. In the `ServiceMeshControlPlane` resource set `global.mtls.enabled` to `true` and `security.selfSigned` set to `false`. {ProductShortName} reads the certificates and key from the secret-mount files.
```
Comment (JStickler) [inline modules/ossm-security-cert-manage.adoc:31]: "resource" is a big vague here.  Could we say "In the `ServiceMeshControlPlane' configuration" or "In the `ServiceMeshControlPlane' YAML file" instead?

Comment (JStickler) [reply modules/ossm-security-cert-manage.adoc:31]: Also, I feel like there is a missing step(s?) here.  If you're in the GUI you'd need to press the Reload button before the new certs are read.  Not sure what the equivalent oc command would be.

File: modules/ossm-security-cert-manage.adoc [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+----
+$ oc delete secret istio.default
+----
+
+[id="ossm-cert-manage-verify-cert"]
+== Verifying your certificates
+
+Use the Bookinfo sample application to verify your certificates are mounted correctly.
+
+1. Retrieve the mounted certificates. Verify the certificates mounted on the pod. Store the pod name in the variable `RATINGSPOD`.
```
Comment (JStickler) [inline modules/ossm-security-cert-manage.adoc:57]: There are three actions in this step, again, I feel like there are missing commands or steps somewhere?  And we're retrieving the certificates in the next step?  And verifying in the last step.  (Is the text in step one really an overview for the whole procedure or what you're doing in step one?)

**Review by JStickler — CHANGES_REQUESTED** [review]
This needs a couple of small tweaks.

File: modules/ossm-security-cert-manage.adoc (hunk 5 above)
Comment (neal-timpe) [reply modules/ossm-security-cert-manage.adoc]: done

File: modules/ossm-security-cert-manage.adoc (hunk 6 above)
Comment (neal-timpe) [reply modules/ossm-security-cert-manage.adoc]: done

File: modules/ossm-security-cert-manage.adoc (hunk 8 above)
Comment (neal-timpe) [reply modules/ossm-security-cert-manage.adoc]: done

File: modules/ossm-security-cert-manage.adoc (hunk 7 above)
Comment (neal-timpe) [reply modules/ossm-security-cert-manage.adoc:31]: @JStickler I left it that way on purpose. It was pointed out to me that you can update the SMCP several ways. I feel like if you're attempting this task, you've already mastered updating the SMCP and you'll do it in the way you've chosen.

However, I'm open to work with you to come up with a framework that we use here, and anywhere we need to update the SMCP. So there's a consistent way. I have some ideas.

File: modules/ossm-security-cert-manage.adoc [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+5. Verify the certificate chain from the root certificate to the workload certificate. Replace `<path>` with the path to your certificates.
++
+----
+$ head -n 21 /tmp/pod-cert-chain.pem > /tmp/pod-cert-chain-workload.pem
+$ openssl verify -CAfile <(cat <path>/ca-cert.pem <path>/root-cert.pem) /tmp/pod-cert-chain-workload.pem
+/tmp/pod-cert-chain-workload.pem: OK
+----
+
+[id="ossm-cert-cleanup"]
+== Cleanup
```
Comment (JStickler) [inline modules/ossm-security-cert-manage.adoc:107]: Can we change this heading to "Removing Certificates" or something more descriptive?

**Review by JStickler — APPROVED** [review]
LGTM
